lib/honeydew.ex

```diff
@@ -196,6 +196,30 @@ defmodule Honeydew do
     |> GenServer.call({:cancel, job})
   end
 
+  @doc """
+  Moves a job to another queue.
+
+  Raises a `RuntimeError` if `to_queue` is not available.
+
+  This function first enqueues the job on `to_queue`, and then tries to
+  cancel it on its current queue. This means there's a possiblity a job could
+  be processed on both queues. This behavior is consistent with Honeydew's
+  at-least-once execution goal.
+
+  This function is most helpful on a queue where there a no workers
+  (like a dead letter queue), because the job won't be processed out from under
+  the queue.
+  """
+  @spec move(Job.t, to_queue :: queue_name) :: Job.t | no_return
+  def move(%Job{} = job, queue) do
+    {:ok, new_job} = enqueue(%Job{job | queue: queue})
```

Comment:
interesting, i've never seen that struct update syntax.

is there an advantage to it over `%{job | queue: queue}`?

Reply:
If you prefix the struct name in the update syntax, then the compiler gives you the warning, instead of waiting until runtime to find out.